tests/test_processing_common.py

```diff
@@ -0,0 +1,169 @@
+# coding=utf-8
+# Copyright 2023 The HuggingFace Inc. team. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import shutil
+
+import numpy as np
+import pytest
+from PIL import Image
+
+from transformers.testing_utils import require_torch
+
+
+@require_torch
+class ProcessorTesterMixin:
+    tokenizer_class = None
+    fast_tokenizer_class = None
+    image_processor_class = None
+    processor_class = None
+
+    def get_tokenizer(self, **kwargs):
+        return self.tokenizer_class.from_pretrained(self.tmpdirname, **kwargs)
+
+    def get_fast_tokenizer(self, **kwargs):
+        return self.fast_tokenizer_class.from_pretrained(self.tmpdirname, **kwargs)
+
+    def get_image_processor(self, **kwargs):
+        return self.image_processor_class.from_pretrained(self.tmpdirname, **kwargs)
+
+    def tearDown(self):
+        shutil.rmtree(self.tmpdirname)
+
+    def prepare_image_inputs(self):
```

Comment:
@ydshieh would be great to get your first feedback on this PR (and potentially your help to extend this to all processors).

I think we may need to create separate `ImageTextProcessorTesterMixin` and `AudioTextProcessorTextMin` classes given that the former use image processor + tokenizer, whereas the latter use feature extractor + tokenizer.

Reply:
I am not in favor of creating 2 classes. Define `prepare_inputs(input_type)`, where `input_type` could be `text`, `image` or `audio`, and the method just returns what is requested.